### PR TITLE
test: added unit tests for initFunctionEntrypoint function

### LINKED ISSUE #395: Cairo runner: add unit tests for `initFunctionEntrypoint` function
Now that #392 is merged, we should add unit tests to confirm validity of our `initFunctionEntrypoint` function inside the Cairo runner.

Related:
- https://github.com/lambdaclass/cairo-vm/blob/4ea224a1068b43aaa3899d24e834475106e8a78f/vm/src/vm/runners/cairo_runner.rs#L1769
- https://github.com/lambdaclass/cairo-vm/blob/4ea224a1068b43aaa3899d24e834475106e8a78f/vm/src/vm/runners/cairo_runner.rs#L1791
- https://github.com/lambdaclass/cairo-vm/blob/4ea224a1068b43aaa3899d24e834475106e8a78f/vm/src/vm/runners/cairo_runner.rs#L1818
- https://github.com/lambdaclass/cairo-vm/blob/4ea224a1068b43aaa3899d24e834475106e8a78f/vm/src/vm/runners/cairo_runner.rs#L4180

### PR DESCRIPTION
**Pull request by Godspower-Eze**
Closes #395

### CODE REVIEW

File: src/vm/runners/cairo_runner.zig [hunk 1]
```diff
@@ -2296,6 +2296,196 @@ test "CairoRunner: initState with no execution_base" {
...
+
+    // Initialize a stack with some values.
+    var stack = ArrayList(MaybeRelocatable).init(std.testing.allocator);
+    defer stack.deinit(); // Deallocate stack memory after the test.
+
+    try stack.append(MaybeRelocatable.fromInt(u8, 7));
+
+    const return_fp = MaybeRelocatable.fromFelt(Felt252.fromInt(u8, 9));
+
+    _ = try cairo_runner.initFunctionEntrypoint(1, return_fp, &stack);
```
Comment (tcoratger) [inline src/vm/runners/cairo_runner.zig:2420]:
```suggestion
    _ = try cairo_runner.initFunctionEntrypoint(1, return_fp, &stack);
        defer cairo_runner.vm.segments.memory.deinitData(std.testing.allocator);
```

File: src/vm/runners/cairo_runner.zig [hunk 2]
```diff
@@ -2293,7 +2293,199 @@ test "CairoRunner: initState with no execution_base" {
...
+
+    const return_fp = MaybeRelocatable.fromFelt(Felt252.fromInt(u8, 9));
+
+    _ = try cairo_runner.initFunctionEntrypoint(0, return_fp, &stack);
+
+    defer cairo_runner.vm.segments.memory.deinitData(std.testing.allocator);
+
+    try expect(cairo_runner.initial_fp.?.eq(cairo_runner.initial_ap.?));
+
+    try expect(cairo_runner.initial_ap.?.eq(Relocatable.init(1, 2)));
```
Comment (tcoratger) [inline src/vm/runners/cairo_runner.zig:2360]: Can you also check that the memory is properly set here. According to the equivalent Rust test, it should be

```rust
        check_memory!(vm.segments.memory, ((1, 0), 9), ((1, 1), (2, 0)));
```

File: src/vm/runners/cairo_runner.zig [hunk 3]
```diff
@@ -2293,7 +2293,199 @@ test "CairoRunner: initState with no execution_base" {
...
+    var vm = try CairoVM.init(std.testing.allocator, .{});
+
+    // Add memory segments to the CairoVM instance.
+    inline for (0..2) |_| _ = try vm.addMemorySegment();
+
+    // Initialize a CairoRunner with an empty program, "plain" layout, and instructions.
+    var cairo_runner = try CairoRunner.init(
+        std.testing.allocator,
+        program,
+        "plain",
```
Comment (tcoratger) [inline src/vm/runners/cairo_runner.zig:2399]: Here same, we should use all cairo layout

File: src/vm/runners/cairo_runner.zig [hunk 4]
```diff
@@ -2293,7 +2293,199 @@ test "CairoRunner: initState with no execution_base" {
...
+
+    const return_fp = MaybeRelocatable.fromFelt(Felt252.fromInt(u8, 9));
+
+    _ = try cairo_runner.initFunctionEntrypoint(1, return_fp, &stack);
+
+    defer cairo_runner.vm.segments.memory.deinitData(std.testing.allocator);
+
+    try expect(cairo_runner.initial_fp.?.eq(cairo_runner.initial_ap.?));
+
+    try expect(cairo_runner.initial_ap.?.eq(Relocatable.init(1, 3)));
```
Comment (tcoratger) [inline src/vm/runners/cairo_runner.zig:2426]: Can you check the memory here too:

```rust
 check_memory!(
            vm.segments.memory,
            ((1, 0), 7),
            ((1, 1), 9),
            ((1, 2), (2, 0))
        );
```

File: src/vm/runners/cairo_runner.zig [hunk 5]
```diff
@@ -2293,7 +2293,199 @@ test "CairoRunner: initState with no execution_base" {
...
+    var vm = try CairoVM.init(std.testing.allocator, .{});
+
+    // Add memory segments to the CairoVM instance.
+    inline for (0..2) |_| _ = try vm.addMemorySegment();
+
+    // Initialize a CairoRunner with an empty program, "plain" layout, and instructions.
+    var cairo_runner = try CairoRunner.init(
+        std.testing.allocator,
+        program,
+        "plain",
```
Comment (tcoratger) [inline src/vm/runners/cairo_runner.zig:2465]: all cairo

**Review by lana-shanghai — APPROVED** [review]
lgtm

**Pull request merged**